File: chat-storage/src/main/java/com/example/chat/storage/domain/message/MessageStatus.java
package com.example.chat.storage.domain.message;

/**
 * 메시지 상태 Enum
 */
public enum MessageStatus {
    /**
     * 대기 중 (발송 전)
     */
    PENDING("pending", "대기 중"),

    /**
     * 발송됨
     */
    SENT("sent", "발송됨"),

    /**
     * 전달됨 (서버가 수신함)
     */
    DELIVERED("delivered", "전달됨"),

    /**
     * 읽음
     */
    READ("read", "읽음"),

    /**
     * 실패
     */
    FAILED("failed", "실패"),

    /**
     * 취소됨
     */
    CANCELLED("cancelled", "취소됨");

    private final String code;
    private final String description;

    MessageStatus(String code, String description) {
        this.code = code;
        this.description = description;
    }

    public String getCode() {
        return code;
    }

    public String getDescription() {
        return description;
    }

    public static MessageStatus fromCode(String code) {
        for (MessageStatus status : values()) {
            if (status.code.equals(code)) {
                return status;
            }
        }
        throw new IllegalArgumentException("Unknown message status code: " + code);
    }

    /**
     * 읽음 상태로 전환 가능한지 확인
     */
    public boolean canTransitionToRead() {
        return this == SENT || this == DELIVERED;
    }

    /**
     * 실패 상태로 전환 가능한지 확인
     */
    public boolean canTransitionToFailed() {
        return this == PENDING || this == SENT;
    }
}

File: apps/chat/libs/chat-domain/src/main/kotlin/com/example/chat/domain/message/MessageStatus.kt
package com.example.chat.domain.message

/**
 * 메시지 상태
 */
enum class MessageStatus {
    /**
     * 대기 중 (아직 발송되지 않음)
     */
    PENDING,

    /**
     * 발송됨 (서버에서 전송 완료)
     */
    SENT,

    /**
     * 전달됨 (수신자에게 도달)
     */
    DELIVERED,

    /**
     * 읽음 (수신자가 읽음)
     */
    READ,

    /**
     * 실패 (발송 실패)
     */
    FAILED
}

